x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/SqlVersion.java

```diff
@@ -23,6 +23,7 @@
     public final byte major;
     public final byte minor;
     public final byte revision;
+    public final byte build; // the build is required for id-compatibility with ES's version
```

Comment:
This is no longer strictly required now TBH.

In a previous version of the PR I had Version turned into an SqlVersion and then back to Version, needed for testing purposes. This resulted in non-equal Version objects, because of this `build` value. The tests no longer require that, but it's still safer to have it like this, IMO.